db/product.go

```diff
@@ -0,0 +1,313 @@
+			return
+		}
+	}
+
+	if count == 0 {
+		err = fmt.Errorf("No records present")
+		logger.WithField("err", err.Error()).Error("No Products were present in database")
+		return
+	}
+
+	// error already handled in product_http
+	ls, _ := strconv.Atoi(limit)
+	ps, _ := strconv.Atoi(page)
+
+	if (count - 1) < (int(ls) * (int(ps) - 1)) {
+		err = fmt.Errorf("Desired Page not found")
+		logger.WithField("err", err.Error()).Error("Page Out Of range")
+		return
+	}
+
+	result, err := s.db.Query(getProductQuery, ls, ps)
+	if err != nil {
+		logger.WithField("err", err.Error()).Error("Error fetching Product Ids from database")
+		return
+	}
+
+	// idArr stores id's of all products
+	var idArr []int
+
+	for result.Next() {
+		var Id int
+		err = result.Scan(&Id)
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Couldn't Scan Product ids")
+			return
+		}
+		idArr = append(idArr, Id)
+	}
+
+	for i := 0; i < len(idArr); i++ {
+		var product Product
+		product, err = s.GetProductByID(ctx, int(idArr[i]))
+		if err != nil {
+			logger.WithField("err", err.Error()).Error("Error selecting Product from database by id " + string(idArr[i]))
+			return
+		}
+		products = append(products, product)
+	}
+
+	return
+}
+
+// CreateNewProduct
+func (s *pgStore) CreateNewProduct(ctx context.Context, p Product) (createdProduct Product, err error) {
+	// First, make sure Product isn't already in db, if Product is present, just return the it
+	err = s.db.Get(&createdProduct, getProductByNameQuery, p.Name)
+	if err == nil {
+		// If there's already a product, err wil be nil, so no new Product is populated.
+		err = fmt.Errorf("Product Already exists!")
+		return
+	}
+	tx, err := s.db.Beginx() // Use Beginx instead of MustBegin so process doesn't die if there is an error
+	if err != nil {
+		// FAIL : Could not begin database transaction
+		logger.WithField("err", err.Error()).Error("Error beginning product insert transaction in db.CreateNewProduct with Id: " + string(p.Id))
+		return
+	}
+
+	_, err = tx.NamedExec(insertProductQuery, p)
+	//  p.Name, p.Description, p.Price, p.Discount, p.Tax, p.Quantity, p.CategoryId, p.Brand, p.Color, p.Size
+
+	if err != nil {
+		// FAIL : Could not run insert Query
+		logger.WithField("err", err.Error()).Error("Error inserting product to database: " + p.Name)
+		return
+	}
+	err = tx.Commit()
+	if err != nil {
+		// FAIL : transaction commit failed.Will Automatically rollback
+		logger.WithField("err", err.Error()).Error("Error commiting transaction inserting product into database: " + string(p.Id))
+		return
+	}
+
+	//length of url
+	urls := len(p.URLs)
+	var number int
+	//new insert record get id number
+	result, err := s.db.Query(newInsertRecord)
+	for result.Next() {
+		err = result.Scan(&number)
+	}
+
+	for i := 0; i < urls; i++ {
+		//insert urls of given records
+		_, err = s.db.Exec(insertProductURLsQuery, number, p.URLs[i])
```

Comment:
we can simply put urls array in product table

Reply:
URLs are no longer separate,  they are included in Product as per your suggestion.